### DISCUSSION TITLE
[Feature] Upload/UpDate sidecar file using an API call

### DISCUSSION THREAD

**Discussion by simulot** [immich-app/immich#3880] [Feature Request]
### The feature

There is a need for modifying asset's metadata, either because the original file didn't provide them, or because the extraction as failed.

Ideally, modifying the metadata either from the web page or using an API won't be available soon.

# workarounds are tedious: 

### reopload the asset
- download and delete the photo
- edit the exif data locally or create a sidecar file
- reopload it.

What will happen to :
- Faces 
- Album 

### placing a sidecar file directly on the server

This suppose the end user has access to the server. I would prefer never touch the server's file system. I won't let my users doing so as well.

# proposal: allow to add a sidecar to an existing asset

### From the web application:
- On the image page, add a button to upload the sidecar file
- Add the possibility to attach a sidecar file to the current photo

### By an API

I know it's possible to upload an asset and its sidecar file by calling the uploadFile end point (https://immich.app/docs/api/upload-file). But this works only once. Each subsequent call result in a duplicate. So the asset must be deleted, re-uploaded with its sidecar file.

- add/modify an API end point for updating the sidecar file

# Use cases

## Geotaging photos en masse
I have a big load of photos taken with digital camera not having a GPS module. 
Some time, metadata aren't correctly extracted (#3875), either GPS position or date.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #3796/#1696
